crates/uv-client/src/registry_client.rs

```diff
@@ -553,14 +557,24 @@ impl RegistryClient {
         &self,
         package_name: &PackageName,
         url: &DisplaySafeUrl,
+        index: &IndexUrl,
         cache_entry: &CacheEntry,
         cache_control: CacheControl<'_>,
     ) -> Result<OwnedArchive<SimpleMetadata>, Error> {
+        let accept = if self
+            .pyx_token_store
+            .as_ref()
+            .is_some_and(|token_store| token_store.is_known_url(index.url()))
+        {
+            MediaType::all()
+        } else {
+            MediaType::pypi()
+        };
```

Comment:
In theory, we should be able to pass `MediaType::all()` to all registries, and registries that don't support it should just ignore the new media types. But I'm worried that misconfigured registries could reject them and error.